octoprint_mrbeam/analytics/analytics_handler.py

```diff
@@ -774,49 +800,54 @@ def _add_other_plugin_data(self, event, event_payload):
         )
 
-    def _add_job_event(self, event, payload=None):
-        self._add_event_to_queue(ak.EventType.JOB, event, payload=payload)
+    def _add_job_event(self, event, payload=None, header_extension=None):
+        self._add_event_to_queue(AnalyticsKeys.EventType.JOB, event, payload=payload, header_extension=header_extension)
 
-    def _add_connectivity_event(self, event, payload):
-        self._add_event_to_queue(ak.EventType.CONNECTIVITY, event, payload=payload)
+    def _add_connectivity_event(self, event, payload, header_extension=None):
+        self._add_event_to_queue(
+            AnalyticsKeys.EventType.CONNECTIVITY, event, payload=payload, header_extension=header_extension)
 
-    def _add_event_to_queue(self, event_type, event_name, payload=None, analytics=True):
+    def _add_event_to_queue(self, event_type, event_name, payload=None, header_extension=None, analytics=True):
         try:
             data = dict()
             if isinstance(payload, dict):
                 data = payload
 
+            if not isinstance(header_extension, dict):
+                header_extension = dict()
+
             event = {
-                ak.Header.SNR: self._snr,
-                ak.Header.TYPE: event_type,
-                ak.Header.ENV: self._plugin.get_env(),
-                ak.Header.VERSION: self.ANALYTICS_LOG_VERSION,
-                ak.Header.EVENT: event_name,
-                ak.Header.TIMESTAMP: time.time(),
-                ak.Header.NTP_SYNCED: self._plugin.is_time_ntp_synced(),
-                ak.Header.SESSION_ID: self._session_id,
-                ak.Header.VERSION_MRBEAM_PLUGIN: self._plugin_version,
-                ak.Header.SOFTWARE_TIER: self._settings.get(["dev", "software_tier"]),
-                ak.Header.DATA: data,
-                ak.Header.UPTIME: get_uptime(),
-                ak.Header.MODEL: self._plugin.get_model_id(),
+                AnalyticsKeys.Header.SNR: self._snr,
+                AnalyticsKeys.Header.TYPE: event_type,
+                AnalyticsKeys.Header.ENV: self._plugin.get_env(),
+                AnalyticsKeys.Header.VERSION: self.ANALYTICS_LOG_VERSION,
+                AnalyticsKeys.Header.EVENT: event_name,
+                AnalyticsKeys.Header.TIMESTAMP: time.time(),
+                AnalyticsKeys.Header.NTP_SYNCED: self._plugin.is_time_ntp_synced(),
+                AnalyticsKeys.Header.SESSION_ID: self._session_id,
+                AnalyticsKeys.Header.VERSION_MRBEAM_PLUGIN: self._plugin_version,
+                AnalyticsKeys.Header.SOFTWARE_TIER: self._settings.get(["dev", "software_tier"]),
+                AnalyticsKeys.Header.DATA: data,
+                AnalyticsKeys.Header.UPTIME: get_uptime(),
+                AnalyticsKeys.Header.MODEL: self._plugin.get_model_id(),
+                AnalyticsKeys.Header.FEATURE_ID: header_extension.get(AnalyticsKeys.Header.FEATURE_ID)
```

Comment:
```suggestion
                AnalyticsKeys.Header.FEATURE_ID: header_extension.get(AnalyticsKeys.Header.FEATURE_ID, None)
```
I believe it is more readable this way. What do you think?